packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+    // Since things like the bonded amount can change out from under us,
+    // we shouldn't rely on it completely.
+    // TODO: consider performance implications vs. diagnostics.
+    await checkAccountState(address, lienDelta);
+
+    const current = getOrElse(amountByAddress, address, () => empty);
+    const target = add(current, lienDelta);
+
+    // Since our mint is not subject to normal supply/demand,
+    // minting here and failing in setLiened below does no harm.
+    const attestationPaymentP = mintZCFMintPayment(
+      zcf,
+      zcfMint,
+      wrapLienedAmount(address, lienDelta),
+    );
+
+    // If setLiened returns successfully, then
+    // the preconditions were indeed met and we
+    // can update the stored liened amount.
+    await E(lienBridge).setLiened(address, current, target);
+    amountByAddress.set(address, target);
+    return attestationPaymentP;
+  };
+
+  /** @param {ZCFSeat} seat */
+  const returnAttestation = async seat => {
+    assertProposalShape(seat, { give: { Attestation: null }, want: {} });
+
+    /** @type {Amount<'copyBag'>} */
+    const attestationAmount = seat.getAmountAllocated(KW.Attestation);
+    const { address, lienedAmount: amountReturned } =
+      unwrapLienedAmount(attestationAmount);
+
+    const liened = amountByAddress.get(address); // or throw
+    const updated = subtract(liened, amountReturned); // or throw
+
+    await E(lienBridge).setLiened(address, liened, updated);
+    // COMMIT. Burn the escrowed attestation payment and exit the seat
+    amountByAddress.set(address, updated);
+    zcfMint.burnLosses(harden({ Attestation: attestationAmount }), seat);
+    seat.exit();
+  };
+
+  /**
+   * Get liened amount synchronously.
+   *
+   * @param {string} address
+   * @param {Brand<'nat'>} brand
+   */
+  const getLiened = (address, brand) => {
+    assert.typeof(address, 'string');
+    assert.equal(
+      brand,
+      stakeBrand,
+      X`This contract can only make attestations for ${brand}`,
+    );
+    return getOrElse(amountByAddress, address, () => empty);
+  };
+
+  const lienMint = harden({
+    mintAttestation,
+    returnAttestation,
+    getLiened,
+    /** @param { Amount<'copyBag'> } attAmt */
+    unwrapLienedAmount: attAmt => unwrapLienedAmount(attAmt).lienedAmount,
+  });
+  return harden({ issuer, brand: attBrand, lienMint });
+};
+
+/**
+ * Authorize each account holder to lien some of their staked assets
+ * and get an attestation that the lien is in place.
+ *
+ * @param {ZCF} zcf
+ * @param {Brand} stakeBrand
+ * @param {string} keyword for attestation issuer
+ * @param {ERef<StakingAuthority>} lienBridge
+ *
+ * NOTE: the liened amount is kept both here in JS and on the
+ * other side of the lienBridge. The mint created here
+ * is the authority on the liened amount. The liened amount
+ * on the other side of the bridge may lag. Do not create two
+ * mints that share a bridge.
```

Comment:
Is there a way to automatically detect this?

Reply:
Sort of... the `lienBridge` could have a pointer back to the calling object and throw if it's already used. But it's a bit of whack-a-mole: we would have the same prohibition not to make two `lienBridge`s. I think the complexity adds more risk than it's worth, given that the callers are all in extremely privileged bootstrap code.

Reply:
Makes sense not to complicate the code. Is there a way with telemetry to see if this problem does occurr?